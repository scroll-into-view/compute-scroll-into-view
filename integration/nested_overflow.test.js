beforeAll(async () => {
  await page.goto('http://localhost:3000/integration/nested_overflow')
})

describe('scrollMode: if-needed', () => {
  test('target in view, container out of view', async () => {
    const actual = await page.evaluate(() => {
      const container = document.querySelector('.container')
      window.scrollTo(0, 0)
      container.scrollTo(250, 250)
      return window.computeScrollIntoView(document.querySelector('.target'), {
        scrollMode: 'if-needed',
      })
    })
    expect(actual).toHaveLength(2)
    expect(actual).toMatchSnapshot()
  })
})
