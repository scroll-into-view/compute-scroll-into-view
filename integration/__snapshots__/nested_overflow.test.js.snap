// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`scrollMode: if-needed target in view, container out of view 1`] = `
Array [
  Object {
    "el": "div.container",
    "left": 250,
    "top": 250,
  },
  Object {
    "el": "html",
    "left": 1250,
    "top": 1001,
  },
]
`;
